sdk/core/System.ClientModel/src/Convenience/ClientConnection.cs

```diff
@@ -69,16 +69,27 @@ internal ClientConnection(string id, string locator, CredentialKind credentialKi
     /// Initializes a new instance of the <see cref="ClientConnection"/> struct with the specified subclient ID.
     /// It is only for the JSON serializer.
     /// </summary>
-    /// <param name="metadata">The connection metadata.</param>
     /// <param name="id">The identifier for the connection.</param>
     /// <param name="locator">The endpoint or resource identifier.</param>
+    /// <param name="credential">The client credential.</param>
     /// <param name="credentialKind">The kind of connection used by the client</param>
-    internal ClientConnection(IReadOnlyDictionary<string, string> metadata, string id, string locator, CredentialKind credentialKind)
+    /// <param name="metadata">The connection metadata.</param>
+    public ClientConnection(string id, string locator, object credential, CredentialKind credentialKind, IReadOnlyDictionary<string, string>? metadata)
     {
+        if (string.IsNullOrWhiteSpace(id))
```

Comment:
Please use braces for all block statements, even if there's a single line in the block.